app/src/main/java/mozilla/lockbox/adapter/SettingCellConfiguration.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.adapter
+
+abstract class SettingCellConfiguration(
+    open val title: String,
+    open val subtitle: String? = null
+)
+
+class TextSettingConfiguration(
+    override val title: String,
+    override val subtitle: String? = null,
+    val detailText: String
+) : SettingCellConfiguration(title, subtitle)
+
+class ToggleSettingConfiguration(
+    override val title: String,
+    override val subtitle: String? = null,
+    val buttonTitle: String? = null,
+    val toggle: Boolean
+) : SettingCellConfiguration(title)
+
+class AppVersionSettingConfiguration(
+    val text: String
+) : SettingCellConfiguration(text)
```

Comment:
A little confused why we didn't use [PreferenceFragmentCompat][1] here.

[1]: https://developer.android.com/reference/android/support/v7/preference/PreferenceFragmentCompat